Expensify/App
# Hide message header when editing

### Details

Thanks for the review! Let's hide the name/date header of messages when in edit mode. This reduces UI clutter and jerky transitions, and brings this behavior in-line with other popular chat clients. #design-approved ✅ 

#### Before
![2022-01-18_20-17-04 (1)](https://user-images.githubusercontent.com/2438855/150077256-0744c43b-7e6c-49a6-a7e0-8218a921bc7d.gif)

#### After
![2022-01-18_20-56-15 (1)](https://user-images.githubusercontent.com/2438855/150077245-a3d33f9d-5c56-4f5a-ab2f-ae4aae4c5be3.gif)

### Fixed Issues
$ https://github.com/Expensify/App/issues/6901

### Tests / QA Steps
1. Open a chat and send a message
2. Hover over the message/tap and hold the message to bring up the options menu and select "Edit comment"
3. Observe that the name/date header of the message disappears when entering edit mode.
4. Save/Cancel the message edit and observe that the header has returned

### Tested On


## iOS Screenshot
![2022-01-19_14-47-48 (1)](https://user-images.githubusercontent.com/2438855/150078797-6bd1e756-9201-458f-8c9a-8e502aa4d665.gif)

## Android Screenshot
![2022-01-20_16-20-18 (1)](https://user-images.githubusercontent.com/2438855/150300143-68f96d49-06d1-4a4c-a125-0771b6ce2f04.gif)

## Desktop Screenshot
![2022-01-20_16-23-07 (1)](https://user-images.githubusercontent.com/2438855/150300523-afb46eb4-3fa3-470a-920e-efd76cf01de4.gif)
